pkg/util/lock/lock.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lock
+
+import (
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/juju/fslock"
+	"github.com/pkg/errors"
+	"k8s.io/minikube/pkg/util/retry"
+)
+
+// WriteWithLock decorates ioutil.WriteFile with a file lock and retry
+func WriteFile(filename string, data []byte, perm os.FileMode) error {
```

Comment:
Do you mind calling log.Infof to say we are about to write to a particular file here? It will come in handy for future debugging.